lib/ui/zoomable_live_image.dart

```diff
@@ -0,0 +1,273 @@
+import 'dart:io' as io;
+import 'package:chewie/chewie.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:logging/logging.dart';
+import 'package:photo_view/photo_view.dart';
+import 'package:photos/core/cache/thumbnail_cache.dart';
+import 'package:photos/core/constants.dart';
+import 'package:photos/core/event_bus.dart';
+import 'package:photos/db/files_db.dart';
+import 'package:photos/events/local_photos_updated_event.dart';
+import 'package:photos/models/file.dart';
+import 'package:photos/ui/loading_widget.dart';
+import 'package:photos/utils/file_util.dart';
+import 'package:photos/utils/thumbnail_util.dart';
+import 'package:photos/utils/toast_util.dart';
+import 'package:video_player/video_player.dart';
+
+class ZoomableLiveImage extends StatefulWidget {
+  final File photo;
+  final Function(bool) shouldDisableScroll;
+  final String tagPrefix;
+  final Decoration backgroundDecoration;
+
+  ZoomableLiveImage(
+    this.photo, {
+    Key key,
+    this.shouldDisableScroll,
+    @required this.tagPrefix,
+    this.backgroundDecoration,
+  }) : super(key: key);
+
+  @override
+  _ZoomableLiveImageState createState() => _ZoomableLiveImageState();
+}
+
+class _ZoomableLiveImageState extends State<ZoomableLiveImage>
```

Comment:
Instead of managing all this state, do you want to use `ZoomableImageView` and `VideoWidget` on top of each other?